### PR TITLE
HPCC-16632 Add cassandra logging agent to init files

### PR DESCRIPTION
**Pull request by wangkx**
Aslo add LogSourceMap to init files for ESP logging agent.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

### CODE REVIEW

**Comment by wangkx** [comment]
@rpastrana this is a 6.2 task. Please review.

**Review by rpastrana — COMMENTED** [review]
@wangkx take a look

File: initfiles/componentfiles/configxml/buildsetCC.xml.in [hunk 1]
```diff
@@ -330,6 +330,12 @@
     </BuildSet>
     <BuildSet deployable="no"
              installSet="deploy_map.xml"
+             name="cassandraloggingagent"
```
Comment (rpastrana) [inline initfiles/componentfiles/configxml/buildsetCC.xml.in]: Make sure we're not adding these components to the Configmanager UI if the entire UI isn't implemented yet.

File: initfiles/componentfiles/configxml/@temp/esp_service_wslogging.xsl [hunk 2]
```diff
@@ -19,56 +38,25 @@ xmlns:seisint="http://seisint.com"  xmlns:set="http://exslt.org/sets" exclude-re
...
-                    <xsl:if test="string($espNetAddress) = ''">
-                        <xsl:message terminate="yes">ESP NetAddress is undefined!</xsl:message>
-                    </xsl:if>
-                    <xsl:variable name="wsloggingUrl"><xsl:text>http://</xsl:text><xsl:value-of select="$espNetAddress"/><xsl:text>:</xsl:text><xsl:value-of select="$espPort"/></xsl:variable>
-                        <LogAgent name="{$agentName}" type="LogAgent" services="GetTransactionSeed,UpdateLog" plugin="espserverloggingagent">
-                            <ESPServer url="{$wsloggingUrl}" user="{$agentNode/@User}" password="{$agentNode/@Password}"/>
-                            <xsl:if test="string($agentNode/@MaxServerWaitingSeconds) != ''">
-                                <MaxServerWaitingSeconds><xsl:value-of select="$agentNode/@MaxServerWaitingSeconds"/></MaxServerWaitingSeconds>
-                            </xsl:if>
+            <xsl:if test="string(CassandraLoggingAgent[1]/@name) = ''">
```
Comment (rpastrana) [inline initfiles/componentfiles/configxml/@temp/esp_service_wslogging.xsl]: Just noticed this test is duplicated in LN... we might want to componetize into individual .xls files and include them into esp_service_wslogging.xsl

**Comment by wangkx** [comment]
@rpastrana please review my changes.

File: esp/logging/logginglib/datafieldmap.cpp [hunk 3]
```diff
@@ -33,7 +33,7 @@ void readLogGroupCfg(IPropertyTree* cfg, StringAttr& defaultLogGroup, MapStringT
     Owned<IPropertyTreeIterator> iter = cfg->getElements("LogGroup");
     ForEach(*iter)
     {
-        ensureInputString(iter->query().queryProp("@name"), true, groupName, -1, "LogGroup @name required");
+        ensureInputString(iter->query().queryProp("@Name"), true, groupName, -1, "LogGroup @Name required");
```
Comment (richardkchapman) [inline esp/logging/logginglib/datafieldmap.cpp]: Rodrigo suggested camel casing for consistency, but NOT leading capitals... please revert this

File: initfiles/componentfiles/configxml/@temp/logging_agent.xsl [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+################################################################################
+-->
+
+<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xml:space="default"
+xmlns:seisint="http://seisint.com"  xmlns:set="http://exslt.org/sets" exclude-result-prefixes="seisint set">
```
Comment (rpastrana) [inline initfiles/componentfiles/configxml/@temp/logging_agent.xsl]: Should we stop using the seisint namespace?

File: initfiles/componentfiles/configxml/@temp/logging_agent.xsl [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+                                </xsl:otherwise>
+                            </xsl:choose>
+                        </xsl:for-each>
+                    </Fieldmap>
+                </xsl:for-each>
+            </LogGroup>
+        </xsl:for-each>
+    </xsl:template>
+
+    <xsl:template name="CassandraLoggingAgent">
```
Comment (rpastrana) [inline initfiles/componentfiles/configxml/@temp/logging_agent.xsl]: should we componentize  specific logging agent type logic?

File: initfiles/componentfiles/configxml/@temp/logging_agent.xsl (hunk 4 above)
Comment (wangkx) [inline initfiles/componentfiles/configxml/@temp/logging_agent.xsl]: I guess I can remove them.

File: initfiles/componentfiles/configxml/@temp/logging_agent.xsl (hunk 5 above)
Comment (wangkx) [inline initfiles/componentfiles/configxml/@temp/logging_agent.xsl]: There is a JIRA for this: https://track.hpccsystems.com/browse/HPCC-16702.

**Review by rpastrana — APPROVED** [review]
@wangkx changes look good, there were a couple of unanswered concerns that might just need an explanation.

File: initfiles/componentfiles/configxml/@temp/esp_service_DynamicESDL.xsl [hunk 6]
```diff
@@ -100,7 +100,7 @@
                         <xsl:message terminate="yes">ESP NetAddress is undefined!</xsl:message>
                     </xsl:if>
                     <xsl:variable name="wsloggingUrl"><xsl:text>http://</xsl:text><xsl:value-of select="$espNetAddress"/><xsl:text>:</xsl:text><xsl:value-of select="$espPort"/></xsl:variable>
-                        <LogAgent name="{$agentName}" type="LogAgent" services="GetTransactionSeed,UpdateLog" plugin="espserverloggingagent">
+                        <LogAgent name="{$agentName}" type="LogAgent" services="GetTransactionSeed,UpdateLog,GetTransactionID" plugin="espserverloggingagent">
```
Comment (wangkx) [inline initfiles/componentfiles/configxml/@temp/esp_service_DynamicESDL.xsl]: The @services attribute includes the functionalities this Logging Agent is designed for.

File: initfiles/componentfiles/configxml/buildsetCC.xml.in (hunk 1 above)
Comment (wangkx) [inline initfiles/componentfiles/configxml/buildsetCC.xml.in]: As we discussed, the existing ConfigMgr UI (can and) should be used to create a  Cassandra based Logging Agent. The part of configurations that the existing ConfigMgr UI does not support (for both  Cassandra based Logging Agent and MySQL based Logging Agent) is the LogGroup and the LogSourceMap. An Red Book item will be added to explain how to add them into environment.xml without using ConfigMgr.

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.

**Pull request merged**